Anlab.Mvc/Client/components/ClientIdModal.tsx

```diff
@@ -0,0 +1,100 @@
+﻿import * as React from 'react';
+import Input from 'react-toolbox/lib/input';
+import { Dialog } from "react-toolbox/lib/dialog";
+import { Button } from "react-toolbox/lib/button";
+import { ClientIdModalInput } from './ClientIdModalInput';
+
+export interface INewClientInfo {
+    employer: string;
+    name: string;
+    email: string;
+    phoneNumber: string;
+}
+
+interface IClientIdModalProps {
+    clientInfo: INewClientInfo;
+    updateClient: Function;
+}
+
+interface IClientIdModalState {
+    newClientInfo: INewClientInfo;
+    active: boolean;
+    isValid: boolean;
+}
+
+export class ClientIdModal extends React.Component<IClientIdModalProps, IClientIdModalState> {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            newClientInfo: {
+                employer: this.props.clientInfo.employer,
+                name: this.props.clientInfo.name,
+                email: this.props.clientInfo.email,
+                phoneNumber: this.props.clientInfo.phoneNumber
+            },
```

Comment:
@laholstege since state.newClientInfo is the same type as props.clientInfo, I think you can make this whole thing into `newClientInfo: ...this.props.clientInfo` which just spreads and copies the properties.

Also just to keep in mind, though it doesn't hurt anything here -- usually copying props into state is not great since if props.clientInfo ever changed, this component would not update to reflect the changes.  Again it's ok here since we know props.clientInfo doesn't change outside of this component, but just something to remember